pkg/plugins/codegen/jenny_plugingotypes.go

```diff
@@ -0,0 +1,61 @@
+package codegen
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/grafana/pkg/plugins/codegen/kindsys"
+)
+
+func PluginGoTypesJenny(root string, inner codejen.OneToOne[*kindsys.PluginDecl]) codejen.OneToOne[*kindsys.PluginDecl] {
+	return &pgoJenny{
+		inner: inner,
+		root:  root,
+	}
+}
+
+type pgoJenny struct {
+	inner codejen.OneToOne[*kindsys.PluginDecl]
+	root  string
+}
+
+func (j *pgoJenny) JennyName() string {
+	return "PluginGoTypesJenny"
+}
+
+func (j *pgoJenny) Generate(decl *kindsys.PluginDecl) (*codejen.File, error) {
+	b := decl.PluginMeta.Backend
+	if b == nil || !*b {
+		return nil, nil
+	}
+
+	buf := new(bytes.Buffer)
+
+	if err := tmpls.Lookup("autogen_header.tmpl").Execute(buf, templateVars_autogen_header{
+		GeneratorPath:  "public/app/plugins/gen.go", // FIXME hardcoding is not OK
+		LineagePath:    filepath.ToSlash(filepath.Join(decl.PluginPath, "models.cue")),
+		LineageCUEPath: decl.Slot.Name(),
+		GenLicense:     true,
+	}); err != nil {
+		return nil, fmt.Errorf("error generating file header: %w", err)
+	}
```

Comment:
i _think_ having this will result in a double-header on any generated Go files?

i made https://gist.github.com/sdboyer/4f79e228fd5c1e9924b970c8ba60510b a while ago, you could use it to test

Reply:
Was on my way changing this with my latest changes but decided to skip since we don't currently have any other Go code being generated. Think easier to do in a follow up PR when we have proper examples of plugins where Go code is generated.

Reply:
yeah fair that WFM